static/js/containers/pages/DashboardPage.js

```diff
@@ -54,6 +58,28 @@ export class DashboardPage extends React.Component<Props, State> {
   }
 
   componentDidMount() {
+    const cookies = new Cookies()
+    const transactionCookie = cookies.get("payload")
+    if (transactionCookie && SETTINGS.gaTrackingID) {
+      try {
+        const transactionObject = JSON.parse(
+          decodeURIComponent(transactionCookie)
+        )
+        // eslint-disable-next-line no-undef
+        dataLayer.push({
+          event:            "purchase",
+          transactionId:    transactionObject.transaction_id,
+          transactionTotal: transactionObject.transaction_total,
+          productType:      transactionObject.product_type,
+          coursewareId:     transactionObject.courseware_id,
+          referenceNumber:  transactionObject.reference_number
+        })
```

Comment:
How did you determine this list of fields to push into GTM?

Reply:
@pdpinch `transactionId`, `transactionTotal` and `referenceNumber` are required for GA e-commerce event. I've also included `coursewareId` and `productType` for sales tracking so that the segmentation can be done on the GA dashboard.